src/app/shared/request/start-request.service.ts

```diff
@@ -1,70 +1,79 @@
 import {Injectable} from '@angular/core';
 import {Request} from "./request.model";
 import {RequestService} from "./request.service";
-import {User} from "../models/user.model";
 import { Grant } from '../models/grant.model';
 import { Advice } from '../models/advice.model';
-import { Question } from '../models/question.model';
-import { Answer } from '../models/answer.model';
+import {AuthService} from "../service/auth.service";
+import {Subscription} from "rxjs";
+import {Admin} from "../models/admin.model";
+import {Answer} from "../models/answer.model";
+import {Question} from "../models/question.model";
 
 @Injectable({providedIn: 'root'})
 export class StartRequestService {
-  //TODO: Remove the part after =
-  currentUser: User = new User(1, "brandon", "password", "yes");
+  currentAdmin: Admin = null;
+  private adminSubscription: Subscription;
 
-  constructor(private requestService: RequestService) {
+  constructor(private requestService: RequestService,
+              private authService: AuthService) {
+    this.findIfRegistered();
   }
 
+  private findIfRegistered() {
+    this.adminSubscription = this.authService.admin.subscribe(admin => {
+      this.currentAdmin = admin;
+    });
+  }
 
   public establishConnection() {
-    let request: Request = new Request(1, "GET", "all", null, false)
+    let request: Request = new Request("", "GET", "all", null, false)
     this.requestService.sendRequestGrant(request)
       .subscribe(response => {
       });
   }
 
   public makeRequestOfGrant(duty: string, specific: string, variables?: Grant) {
-    const request: Request = new Request(this.currentUser.id, duty, specific, variables, false);
+    const request: Request = new Request(this.currentAdmin.token, duty, specific, variables, false);
```

Comment:
Je stuurt nu elke keer een token mee wat niet erg is, maar als je niet ingelogd bent leest hij een propertie van iets wat null is waardoor hij nooit een request maakt.
AKA als ik niet ingelogd ben haalt hij nooit alle vragen op nu.

Reply:
Dta doet die wel dta is getest